File: 2792-neighboring-bitwise-xor/neighboring-bitwise-xor.py
from typing import List

class Solution:
    def doesValidArrayExist(self, derived: List[int]) -> bool:
        n = len(derived)
        
        def is_valid(start_value: int) -> bool:
            original = [start_value]
            for i in range(1, n):
                original.append(derived[i - 1] ^ original[i - 1])
            return original[-1] ^ original[0] == derived[-1]
        
        return is_valid(0) or is_valid(1)

File: 2792-neighboring-bitwise-xor/neighboring-bitwise-xor.c
#include <stdbool.h>

bool doesValidArrayExist(int* derived, int derivedSize) {
    
    bool is_valid(int start_value) {
        int current = start_value;
        for (int i = 0; i < derivedSize - 1; i++) {
            current = derived[i] ^ current;
        }
        return (derived[derivedSize - 1] ^ current) == start_value;
    }
    

    return is_valid(0) || is_valid(1);
}
